lib/datadog/lambda/metrics.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+#
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+#
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2023 Datadog, Inc.
+#
+
+require 'datadog/statsd'
+
+module Datadog
+  # Metrics module contains the singleton class Client to send custom
+  # metrics to Datadog.
+  module Metrics
+    URL = 'localhost'
+    PORT = '8125'
+    # Client is a singleton class that instantiates a Datadog::Statsd
+    # client to send metrics to the Datadog Extension if present.
+    class Client
+      private_class_method :new
+
+      def self.instance
+        @instance ||= new
+      end
+
+      def distribution(name, value, time: nil, **tags)
+        tag_list = get_tags(**tags)
+
+        if Datadog::Utils.extension_running?
+          Datadog::Utils.logger.debug 'sending metrics through extension'
+          begin
+            @statsd.distribution(name, value, tags: tag_list)
+          rescue StandardError => e
+            Datadog::Utils.logger.warning "error sending metric to the extension: #{e}"
+          end
+        else
+          Datadog::Utils.logger.debug 'metrics are going to be handled by the forwarder'
```

Comment:
If this logic gets more complicated in the future (e.g. if we are going to implement other statsd methods like `gauge`, `increment`, `histogram`) I suggest creating two separate subclasses, like ClientStatsd and ClientStdout, and choose which one you want to use at `Client.instance` time:
```ruby
@instance ||= Datadog::Utils.extension_running? ? ClientStatsd.new : ClientStdout.new
```

Reply:
You'd have a base class too, with `def get_tags` for example, as this is shared behavior.

Reply:
Saving this comment and will address in a future PR.

Since other metrics are not on the road map for us at the moment. I'm hesitating to add it as of now.
Will be on top of this suggestion though.